core/src/impl/Kokkos_FunctorAdapter.hpp

```diff
@@ -1334,9 +1334,9 @@ struct FunctorValueTraits<FunctorType, ArgTag,
   using functor_type = FunctorType;
 
   static_assert(
-      IS_VOID || IS_REJECT || 0 == (sizeof(ValueType) % sizeof(int)),
+      IS_VOID || IS_REJECT || sizeof(ValueType) != 3,
```

Comment:
do you actually need that? I think your logic would just use a value_count of 3 and a size_type of int8_t? Might be worthwhile to check with a struct of 3 chars and alignas(3)

Reply:
i will add a test for this.